### PR TITLE
Concept for simplifying get & watch in builds controller

### PR DESCRIPTION
**Pull request by benjaminapetersen**
Related to issue #129.  Attempting to eliminate some copy/paste duplication via reusable resolve & reject functions.

@jwforres @spadgett

### CODE REVIEW

File: app/scripts/controllers/build.js [hunk 1]
```diff
@@ -67,72 +126,17 @@ angular.module('openshiftConsole')
...
-          // success
-          function(build) {
-
-            $scope.loaded = true;
-            $scope.build = build;
-            setLogVars(build);
-            var buildNumber = $filter("annotation")(build, "buildNumber");
-            if (buildNumber) {
-              $scope.breadcrumbs[2].title = "#" + buildNumber;
-            }
```
Comment (benjaminapetersen) [inline app/scripts/controllers/build.js]: The request flow may be easier to follow here, but I have no problem with the inline anonymous functions.  Gain is mostly in DRY.

File: app/scripts/controllers/build.js [hunk 2]
```diff
@@ -67,72 +126,17 @@ angular.module('openshiftConsole')
...
-            buildConfigName = build.metadata.labels.buildconfig;
-            buildName = build.metadata.name;
-          }
+            watches.push(DataService.watchObject("builds", $routeParams.build, context, buildResolved));
+            watches.push(DataService.watchObject("buildconfigs", $routeParams.buildconfig, context, buildConfigResolved));
+          }, buildRejected);
 
-          updateCanBuild();
-        }));
+        watches.push(DataService.watch("builds", context, buildsResolved));
```
Comment (jwforres) [inline app/scripts/controllers/build.js]: actually... are we even using builds for anything anymore, we use to use it in canBuild to disable the Start Build button, but stopped doing that when build serialization went into the API.  I think we can stop watching builds altogether now.

Comment (benjaminapetersen) [inline app/scripts/controllers/build.js]: Oh nice.  Yeah, I didn't even look at that, was just going for a simple ctrl to prototype sharing the get/watch callback but cutting more code is a win! :smile:

File: app/scripts/controllers/build.js [hunk 3]
```diff
@@ -67,72 +126,17 @@ angular.module('openshiftConsole')
...
-            }
-          });
-
-          var buildConfigName;
-          var buildName;
-          if (build) {
-            buildConfigName = build.metadata.labels.buildconfig;
-            buildName = build.metadata.name;
-          }
+            watches.push(DataService.watchObject("builds", $routeParams.build, context, buildResolved));
```
Comment (jwforres) [inline app/scripts/controllers/build.js]: now that we've plumbed the labelSelector / fieldSelector stuff, we should probably update the watchObject in DataService to tack on the fieldSelector for the name of the object so we aren't pulling a bunch of data we don't need, doesn't need to be done in this PR, just thinking out loud :)

Comment (benjaminapetersen) [inline app/scripts/controllers/build.js]: yeah definitely!

**Comment by benjaminapetersen** [comment]
Cool, will do.  I'll do a file here, a file there & slowly work out some more of these over time.

File: app/scripts/controllers/build.js (hunk 3 above)
Comment (benjaminapetersen) [inline app/scripts/controllers/build.js]: I'll create an issue for tracking that.

**Comment by benjaminapetersen** [comment]
Updated & dropped the watch builds

**Comment by benjaminapetersen** [comment]
and rebased. should be set!

**Comment by jwforres** [comment]
[merge]

**Pull request merged**